cocos/3d/assets/material.ts

```diff
@@ -133,6 +139,16 @@ export class Material extends Asset {
         this.update();
     }
 
+    public getHash () {
+        if (!this._dirty) { return this._hash; }
+        let str = '';
+        for (const pass of this._passes) {
+            str += pass.serializePipelineStates();
+        }
+        this._dirty = false;
+        return this._hash = murmurhash2_32_gc(str, 666);
+    }
+
```

Comment:
```suggestion
 public getHash () {
        if (!this._dirty) { return this._hash; }
        let str = '';
        for (const pass of this._passes) {
            str += pass.serializePipelineStates();
        }
        this._dirty = false;
        return this._hash = murmurhash2_32_gc(str, 666);
    }
```

这块有必要用dirty机制吗？在创建完pass后，hash是否就可以确定了